docs/topics/upgrade.md

```diff
@@ -76,6 +76,22 @@ For example,
   --client-secret xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
 ```
 
-### What can go wrong
+## Known Limitations
+
+### Manual reconciliation
+
+The upgrade operation is long running, and for large clusters, more susceptible to single operational failures. This is based on the design principle of upgrade enumerating, one-at-a-time, through each node in the cluster. A transient Azure resource allocation error could thus interrupt the successful progression of the overall transaction. At present, the upgrade operation is implemented to "fail fast"; and so, if a well formed upgrade operation fails before completing, it can be manually retried by invoking the exact same command line arguments as were sent originally. The upgrade operation will enumerate through the cluster nodes, skipping any nodes that have already been upgraded to the desired Kubernetes version. Those nodes that match the *original* Kubernetes version will then, one-at-a-time, be cordon and drained, and upgraded to the desired version. Put another way, an upgrade command is designed to be idempotent across retry scenarios.
+
+### Cluster-autoscaler + VMSS
+
+There are known limitations with VMSS cluster-autoscaler scenarios and upgrade. Our current guidance is not to use `aks-engine upgrade` on clusters with `cluster-autoscaler` functionality. See [here](https://github.com/Azure/aks-engine/issues/400) to get more information and to track progress of the issues related to these limitations.
+
+### Cluster-autoscaler + Availability Set
+
+We don't recommend using `aks-engine upgrade` on clusters that have Availability Set (non-VMSS) agent pools `cluster-autoscaler` at this time.
+
+### Cluster-autoscaler + Dynamic Agent Pools
+
+If you manually add/remove agent pools on your `aks-engine`-created cluster over time, we don't recommend using `aks-engine upgrade` at this time.
```

Comment:
Since we don't support adding/removing agent pools + upgrade in _any_ scenario (with or without autoscaler) I wouldn't point that out as a specific limitation since it might be misleading users to believe that we _do_ support "Dynamic Agent Pools" without cluster-autoscaler.

I think this point is already covered by requirement number 2 "do not upgrade your cluster if you've made changes outside of the upgrade/scale flows".